app/src/main/java/com/github/wnder/MainActivity.java

```diff
@@ -55,25 +55,20 @@ protected void onCreate(Bundle savedInstanceState) {
         imageView.setImageURI(user.getProfilePicture());
 
         //Set the buttons: guess, upload, history
-        findViewById(R.id.getPictureButton).setOnClickListener(this);
-        findViewById(R.id.uploadPictureButton).setOnClickListener(this);
-        findViewById(R.id.menuToHistoryButton).setOnClickListener(this);
+        findViewById(R.id.uploadPictureButton).setOnClickListener(id -> {
+            LocationManager locationManager = (LocationManager) getSystemService(Context.LOCATION_SERVICE);
+            GlobalUser.getUser().setLocation(GlobalUser.getUser().getPositionFromGPS(locationManager, getApplicationContext()));
+            openUploadActivity();
+        });
+
+        findViewById(R.id.getPictureButton).setOnClickListener(id -> openPreviewActivity());
```

Comment:
You should also set the new location when we want to press the getPictureButton, because we want it toimpact the pictures we get.

Reply:
I believe this is no longer necessary?

Reply:
Yes we don't need this anymore so we should remove it.